theseus_gui/src/components/ui/Instance.vue

```diff
@@ -11,23 +17,108 @@ const props = defineProps({
       return {}
     },
   },
+  load: {
+    type: Function,
+    default: () => {},
+  },
 })
+
+const confirmModal = ref(null)
+const stopBtn = ref(null)
+const playBtn = ref(null)
+
+const uuid = ref(null)
+
+const router = useRouter()
+
+const seeInstance = async () => {
+  const instancePath = props.instance.metadata
+    ? `/instance/${encodeURIComponent(props.instance.path)}`
+    : `/project/${encodeURIComponent(props.instance.project_id)}`
+
+  await router.push(instancePath)
+}
+
+const install = async (e) => {
+  e.stopPropagation()
+  props.load()
+  const [data, versions] = await Promise.all([
+    ofetch(
+      `https://api.modrinth.com/v2/project/${
+        props.instance.metadata
+          ? props.instance.metadata?.linked_project_id
+          : props.instance.project_id
+      }`
+    ).then(shallowRef),
+    ofetch(
+      `https://api.modrinth.com/v2/project/${
+        props.instance.metadata
+          ? props.instance.metadata?.linked_project_id
+          : props.instance.project_id
+      }/version`
+    ).then(shallowRef),
+  ])
+
+  if (data.value.project_type === 'modpack') {
+    const packs = Object.values(await list())
+
+    if (
+      packs.length === 0 ||
+      !packs.map((value) => value.metadata).find((pack) => pack.linked_project_id === data.value.id)
+    ) {
+      await pack_install(versions.value[0].id)
+      router.go()
+    } else confirmModal.value.show(versions.value[0].id)
+  }
+  // TODO: Add condition for installing a mod
+}
+
+const play = async (e) => {
+  e.stopPropagation()
+  uuid.value = await run(props.instance.path)
+
+  stopBtn.value.style.opacity = 1
+  stopBtn.value.style.bottom = '4.5rem'
+  stopBtn.value.style.display = 'flex'
+  playBtn.value.style.display = 'none'
+}
+
+const stop = async (e) => {
+  e.stopPropagation()
+  await kill_by_uuid(uuid.value)
+
+  stopBtn.value.style.opacity = 0
+  stopBtn.value.style.bottom = '0'
+  stopBtn.value.style.display = 'none'
+  playBtn.value.style.display = 'flex'
+}
```

Comment:
- Code-wise, seems like a very weird way to handle all of this. This should just set variables and be handled in the DOM
- After stopping an instance, it keeps the stop button